fix(plugin-keychain-vault): no method to_raw found for struct Body

**Describe the bug**

Merging #2084 broke the container build for the keychain vault plugin. 

**To Reproduce**

Try to build the container and watch it fail.

**Expected behavior**

Build should succeed.

**Logs/Stack traces**

[ghcr-keychain-vault-compilation-error-ci.log](https://github.com/hyperledger/cactus/files/9129075/ghcr-keychain-vault-compilation-error-ci.log)

```sh
2022-07-18T00:40:39.7187799Z #17 200.1    Compiling openapi_client v0.3.0 (/cactus-keychain-vault-server)
2022-07-18T00:40:40.1589801Z #17 200.6 error[E0599]: no method named `to_raw` found for struct `Body` in the current scope
2022-07-18T00:40:40.1590651Z #17 200.6    --> src/server/mod.rs:150:35
2022-07-18T00:40:40.1590937Z #17 200.6     |
2022-07-18T00:40:40.1591230Z #17 200.6 150 |                 let result = body.to_raw().await;
2022-07-18T00:40:40.1591550Z #17 200.6     |                                   ^^^^^^ method not found in `Body`
2022-07-18T00:40:40.1591834Z #17 200.6     | 
2022-07-18T00:40:40.1592464Z #17 200.6    ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.8/src/body/body.rs:36:1
2022-07-18T00:40:40.1593097Z #17 200.6     |
2022-07-18T00:40:40.1593340Z #17 200.6 36  | pub struct Body {
2022-07-18T00:40:40.1593673Z #17 200.6     | ---------------
2022-07-18T00:40:40.1594037Z #17 200.6     | |
2022-07-18T00:40:40.1594554Z #17 200.6     | doesn't satisfy `<_ as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1594991Z #17 200.6     | doesn't satisfy `Body: BodyExt`
2022-07-18T00:40:40.1595345Z #17 200.6     | doesn't satisfy `Body: Stream`
2022-07-18T00:40:40.1595616Z #17 200.6     |
2022-07-18T00:40:40.1595949Z #17 200.6     = note: the method `to_raw` exists but the following trait bounds were not satisfied:
2022-07-18T00:40:40.1596354Z #17 200.6             `<Body as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1596707Z #17 200.6             which is required by `Body: BodyExt`
2022-07-18T00:40:40.1596999Z #17 200.6             `Body: Stream`
2022-07-18T00:40:40.1597285Z #17 200.6             which is required by `Body: BodyExt`
2022-07-18T00:40:40.1597642Z #17 200.6             `<&Body as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1597998Z #17 200.6             which is required by `&Body: BodyExt`
2022-07-18T00:40:40.1598273Z #17 200.6             `&Body: Stream`
2022-07-18T00:40:40.1598569Z #17 200.6             which is required by `&Body: BodyExt`
2022-07-18T00:40:40.1598925Z #17 200.6             `<&mut Body as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1599283Z #17 200.6             which is required by `&mut Body: BodyExt`
2022-07-18T00:40:40.1599566Z #17 200.6             `&mut Body: Stream`
2022-07-18T00:40:40.1599868Z #17 200.6             which is required by `&mut Body: BodyExt`
2022-07-18T00:40:40.1600141Z #17 200.6 
2022-07-18T00:40:40.1600449Z #17 200.6 error[E0599]: no method named `to_raw` found for struct `Body` in the current scope
2022-07-18T00:40:40.1600859Z #17 200.6    --> src/server/mod.rs:246:35
2022-07-18T00:40:40.1601126Z #17 200.6     |
2022-07-18T00:40:40.1601406Z #17 200.6 246 |                 let result = body.to_raw().await;
2022-07-18T00:40:40.1601717Z #17 200.6     |                                   ^^^^^^ method not found in `Body`
2022-07-18T00:40:40.1602000Z #17 200.6     | 
2022-07-18T00:40:40.1602487Z #17 200.6    ::: /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.8/src/body/body.rs:36:1
2022-07-18T00:40:40.1602826Z #17 200.6     |
2022-07-18T00:40:40.1603070Z #17 200.6 36  | pub struct Body {
2022-07-18T00:40:40.1603391Z #17 200.6     | ---------------
2022-07-18T00:40:40.1603646Z #17 200.6     | |
2022-07-18T00:40:40.1604124Z #17 200.6     | doesn't satisfy `<_ as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1604682Z #17 200.6     | doesn't satisfy `Body: BodyExt`
2022-07-18T00:40:40.1605033Z #17 200.6     | doesn't satisfy `Body: Stream`
2022-07-18T00:40:40.1605304Z #17 200.6     |
2022-07-18T00:40:40.1605622Z #17 200.6     = note: the method `to_raw` exists but the following trait bounds were not satisfied:
2022-07-18T00:40:40.1606031Z #17 200.6             `<Body as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1606390Z #17 200.6             which is required by `Body: BodyExt`
2022-07-18T00:40:40.1606679Z #17 200.6             `Body: Stream`
2022-07-18T00:40:40.1606970Z #17 200.6             which is required by `Body: BodyExt`
2022-07-18T00:40:40.1607327Z #17 200.6             `<&Body as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1607663Z #17 200.6             which is required by `&Body: BodyExt`
2022-07-18T00:40:40.1607940Z #17 200.6             `&Body: Stream`
2022-07-18T00:40:40.1608233Z #17 200.6             which is required by `&Body: BodyExt`
2022-07-18T00:40:40.1608594Z #17 200.6             `<&mut Body as Stream>::Item = std::result::Result<bytes::bytes::Bytes, _>`
2022-07-18T00:40:40.1609104Z #17 200.6             which is required by `&mut Body: BodyExt`
2022-07-18T00:40:40.1609402Z #17 200.6             `&mut Body: Stream`
2022-07-18T00:40:40.1609704Z #17 200.6             which is required by `&mut Body: BodyExt`
2022-07-18T00:40:40.1610040Z #17 200.6 
2022-07-18T00:40:40.1610298Z #17 200.7 error: aborting due to 2 previous errors
2022-07-18T00:40:40.3099927Z #17 200.7 
2022-07-18T00:40:40.3100582Z #17 200.7 For more information about this error, try `rustc --explain E0599`.
2022-07-18T00:40:40.3100887Z #17 200.7 error: could not compile `openapi_client`
2022-07-18T00:40:40.3101088Z #17 200.7 
2022-07-18T00:40:40.3101393Z #17 200.7 To learn more, run the command again with --verbose.
2022-07-18T00:40:40.3101683Z #17 200.7 warning: build failed, waiting for other jobs to finish...
2022-07-18T00:40:50.3326421Z #17 210.8 error: build failed
2022-07-18T00:40:50.3327524Z #17 ERROR: executor failed running [/bin/sh -c cargo build --release --example server]: exit code: 101
2022-07-18T00:40:50.3465773Z ------
2022-07-18T00:40:50.3466502Z  > [builder 5/6] RUN cargo build --release --example server:
2022-07-18T00:40:50.3468001Z ------
2022-07-18T00:40:50.3468806Z executor failed running [/bin/sh -c cargo build --release --example server]: exit code: 101
2022-07-18T00:40:50.3496406Z ##[error]Process completed with exit code 1.
```

**Cloud provider or hardware configuration:**
Are you running the software on a dev machine or somewhere in the cloud?

**Operating system name, version, build:**

Use this command: `printf "$(uname -srm)\n$(cat /etc/os-release)\n"`

For example

Ubuntu-20.04

**Hyperledger Cactus release version or commit (git rev-parse --short HEAD):**

`main`

**Hyperledger Cactus Plugins/Connectors Used**

Keychain vault plugin

**Additional context**

The change that broke the build was the bumping of a dependency. We need to do it because of security vulnerability issues.

--- a/packages/cactus-plugin-keychain-vault/src/cactus-keychain-vault-server/Dockerfile
+++ b/packages/cactus-plugin-keychain-vault/src/cactus-keychain-vault-server/Dockerfile
@@ -1,4 +1,4 @@
-FROM rust:1.49.0 as builder
+FROM rust:1.63.0 as builder
 
 WORKDIR /
 RUN USER=root cargo new --bin cactus-keychain-vault-server

--- a/packages/cactus-plugin-keychain-vault/src/cactus-keychain-vault-server/rust/gen/Cargo.toml
+++ b/packages/cactus-plugin-keychain-vault/src/cactus-keychain-vault-server/rust/gen/Cargo.toml
@@ -16,10 +16,10 @@ conversion = ["frunk", "frunk_derives", "frunk_core", "frunk-enum-core", "frunk-
 
 [target.'cfg(any(target_os = "macos", target_os = "windows", target_os = "ios"))'.dependencies]
 native-tls = { version = "0.2", optional = true }
-hyper-tls = { version = "0.4", optional = true }
+hyper-tls = { version = "0.5", optional = true }
 
 [target.'cfg(not(any(target_os = "macos", target_os = "windows", target_os = "ios")))'.dependencies]
-hyper-openssl = { version = "0.8", optional = true }
+hyper-openssl = { version = "0.9", optional = true }
 openssl = {version = "0.10", optional = true }
 
 [dependencies]
@@ -27,7 +27,7 @@ openssl = {version = "0.10", optional = true }
 async-trait = "0.1.24"
 chrono = { version = "0.4", features = ["serde"] }
 futures = "0.3"
-swagger = "5.0.0-alpha-1"
+swagger = { version = "6.1", features = ["serdejson", "server", "client", "tls", "tcp"] }
 log = "0.4.0"
 mime = "0.3"
 
@@ -37,7 +37,7 @@ serde_json = "1.0"
 # Crates included if required by the API definition
 
 # Common between server and client features
-hyper = {version = "0.14", optional = true}
+hyper = {version = "0.14", features = ["full"], optional = true}
 serde_ignored = {version = "0.1.1", optional = true}
 url = {version = "2.1", optional = true}
 
@@ -59,12 +59,12 @@ frunk-enum-core = { version = "0.2.0", optional = true }
 [dev-dependencies]
 clap = "2.25"
 env_logger = "0.7"
-tokio = { version = "0.2", features = ["rt-threaded", "macros", "stream"] }
+tokio = { version = "1.14", features = ["full"] }
 native-tls = "0.2"
 tokio-tls = "0.3"
 
 [target.'cfg(not(any(target_os = "macos", target_os = "windows", target_os = "ios")))'.dev-dependencies]
-tokio-openssl = "0.4"
+tokio-openssl = "0.6"
 openssl = "0.10"
 
 [[example]]

--- a/packages/cactus-plugin-keychain-vault/src/cactus-keychain-vault-server/rust/gen/src/server/mod.rs
+++ b/packages/cactus-plugin-keychain-vault/src/cactus-keychain-vault-server/rust/gen/src/server/mod.rs
@@ -147,7 +147,7 @@ impl<T, C> hyper::service::Service<(Request<Body>, C)> for Service<T, C> where
                 // Body parameters (note that non-required body parameters will ignore garbage
                 // values, rather than causing a 400 response). Produce warning header and logs for
                 // any unused fields.
-                let result = body.to_raw().await;
+                let result = hyper::body::to_bytes(body).await;
                 match result {
                             Ok(body) => {
                                 let mut unused_elements = Vec::new();
@@ -243,7 +243,7 @@ impl<T, C> hyper::service::Service<(Request<Body>, C)> for Service<T, C> where
                 // Body parameters (note that non-required body parameters will ignore garbage
                 // values, rather than causing a 400 response). Produce warning header and logs for
                 // any unused fields.
-                let result = body.to_raw().await;
+                let result = hyper::body::to_bytes(body).await;
                 match result {
                             Ok(body) => {
                                 let mut unused_elements = Vec::new();
@@ -342,14 +342,14 @@ impl<T, C> hyper::service::Service<(Request<Body>, C)> for Service<T, C> where
 /// Request parser for `Api`.
 pub struct ApiRequestParser;
 impl<T> RequestParser<T> for ApiRequestParser {
-    fn parse_operation_id(request: &Request<T>) -> Result<&'static str, ()> {
+    fn parse_operation_id(request: &Request<T>) -> Option<&'static str> {
         let path = paths::GLOBAL_REGEX_SET.matches(request.uri().path());
         match request.method() {
             // GetKeychainEntry - POST /api/v1/plugins/@hyperledger/cactus-plugin-keychain-vault/get-keychain-entry
-            &hyper::Method::POST if path.matched(paths::ID_API_V1_PLUGINS_HYPERLEDGER_CACTUS_PLUGIN_KEYCHAIN_VAULT_GET_KEYCHAIN_ENTRY) => Ok("GetKeychainEntry"),
+            &hyper::Method::POST if path.matched(paths::ID_API_V1_PLUGINS_HYPERLEDGER_CACTUS_PLUGIN_KEYCHAIN_VAULT_GET_KEYCHAIN_ENTRY) => Some("GetKeychainEntry"),
             // SetKeychainEntry - POST /api/v1/plugins/@hyperledger/cactus-plugin-keychain-vault/set-keychain-entry
-            &hyper::Method::POST if path.matched(paths::ID_API_V1_PLUGINS_HYPERLEDGER_CACTUS_PLUGIN_KEYCHAIN_VAULT_SET_KEYCHAIN_ENTRY) => Ok("SetKeychainEntry"),
-            _ => Err(()),
+            &hyper::Method::POST if path.matched(paths::ID_API_V1_PLUGINS_HYPERLEDGER_CACTUS_PLUGIN_KEYCHAIN_VAULT_SET_KEYCHAIN_ENTRY) => Some("SetKeychainEntry"),
+            _ => None,
         }
     }
 }

